pkg/apis/kubeflow.org/v2alpha1/trainingruntime_types.go

```diff
@@ -15,3 +15,197 @@ limitations under the License.
+
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired TrainingRuntime.
+	Spec TrainingRuntimeSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// TrainingRuntimeList is a collection of training runtimes.
+type TrainingRuntimeList struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard list metadata.
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// List of TrainingRuntimes.
+	Items []TrainingRuntime `json:"items"`
+}
+
+// TrainingRuntimeSpec represents a specification of the desired training runtime.
+type TrainingRuntimeSpec struct {
+
+	// Configuration for the model training with ML-specific parameters.
+	MLSpec *MLSpec `json:"mlSpec,omitempty"`
+
+	// Configuration for the PodGroup to enable gang-scheduling via supported plugins.
+	PodGroupSpec *PodGroupSpec `json:"podGroupSpec,omitempty"`
+
+	// JobSet template which will be used by TrainJob.
+	Template JobSetTemplateSpec `json:"template"`
+}
+
+// JobSetTemplateSpec represents a template of the desired JobSet.
+type JobSetTemplateSpec struct {
+	// Metadata for custom JobSet's labels and annotations.
+	// JobSet name and namespace is equal to the TrainJob's name and namespace.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired JobSet which will be created from TrainJob.
+	Spec jobsetv1alpha2.JobSetSpec `json:"spec,omitempty"`
+}
+
+// PodGroupSpec represents a PodGroup configuration to enable gang-scheduling.
+type PodGroupSpec struct {
+	// Plugin for the gang-scheduling.
+	Plugin GangSchedulerPlugin `json:"plugin"`
+
+	// Time threshold to schedule PodGroup for gang-scheduling.
+	ScheduleTimeoutSeconds *string `json:"scheduleTimeoutSeconds,omitempty"`
+}
+
+// GangSchedulerPlugin represents one of the supported gang-scheduling plugins.
+type GangSchedulerPlugin string
+
+const (
+	// Volcano plugin for gang-scheduling.
+	GangSchedulerPluginVolcano GangSchedulerPlugin = "volcano"
+
+	// Coscheduling plugin from the Kubernetes scheduler-plugins for gang-scheduling.
+	GangSchedulerPluginCoscheduling GangSchedulerPlugin = "coscheduling"
+)
+
+// MLSpec represents configuration for the model trining with ML-specific parameters.
+type MLSpec struct {
+
+	// Number of training nodes.
+	// Defaults to 1.
+	NumNodes *int32 `json:"numNodes,omitempty"`
+
+	// Configuration for the runtime-specific parameters, such as Torch or MPI.
+	// One of the following spec sources can be set.
+	MLSpecSource `json:",inline"`
+}
+
+// MLPolicySource represents the runtime-specific configuration for various technologies.
+// One of the following specs can be set.
+type MLSpecSource struct {
+
+	// Configuration for the PyTorch runtime.
+	Torch *TorchMLSpecSource `json:"torch,omitempty"`
+
+	// Configuration for the MPI Runtime.
+	MPI *MPIMLSpecSource `json:"mpi,omitempty"`
+}
+
+// TorchMLSpecSource represents a PyTorch runtime configuration.
+type TorchMLSpecSource struct {
```

Comment:
@tenzen-y I removed `standalone` from TorchMLSpecSource API.
I think, we can detect when `numNodes=1` and `numProcPerNode>1` to set the `standalone` parameter which sets  [the default values for rendezvous](https://github.com/pytorch/pytorch/blob/main/torch/distributed/run.py#L893-L897).

Reply:
When users will need it, we can add it in the future.